src/pages/patientView/PatientViewPage.tsx

```diff
@@ -544,6 +552,72 @@ export class PatientViewPageInner extends React.Component<
         return this.patientViewPageStore;
     }
 
+    @action.bound
+    private handleNavigationControlClick() {
+        const patientIdentifiers = this.pageStore.patientIdsInCohort.map(p => {
+            const patientIdParts = p.split(':');
+            return {
+                patientId: patientIdParts[1],
+                studyId: patientIdParts[0],
+            };
+        });
+        const queriedStudies: Set<string> = new Set(
+            patientIdentifiers.map(p => p.studyId)
+        );
+
+        const studyPage = window.open(
+            buildCBioPortalPageUrl(`study`, {
+                id: Array.from(queriedStudies).join(','),
+            }),
+            '_blank'
+        );
+        if (patientIdentifiers.length > 0) {
+            (studyPage as any).studyPageFilter = `filterJson=${JSON.stringify({
+                patientIdentifiers,
+            })}`;
+        }
+    }
+
+    @action.bound
+    private handleDeletePatient(deleteAtIndex: number) {
+        this.pageStore.patientIdsInCohort.splice(deleteAtIndex, 1);
```

Comment:
it's odd that you essentially mutate the state on this first line, then later use window.open to arrive at the new url.  better to use a functional method (lodash) to get a new array (without mutating old one in place).   but it also begs question, why do we need to use window.open at all down below?

Reply:
Yes, without doing `window.open` url won't be getting updated (with `URLWrapper` we could only update params not hash)

Reply:
You could use urlWrapper to update hash.   urlWrapper uses the underlying ExtendedRouterStore (src/shared/lib/ExtendedRouterStore.ts) line 97, you can see we are just passing through the existing hash.  but this could easily be adapted to accept a new hash.